### PR TITLE
Extra parameter to start a container

### PR DESCRIPTION
**Pull request by caguero**
# 🦟 Bug fix

Leaving it as a draft to discuss what people think. The idea behind this extra parameter is to provide more flexibility. We'd have three options:

1. `use_composition:=False` . Run as an external ROS node.
2. `use_composition:=True` and `create_own_container`. Load as a composable node within its own ROS container.
3. `use_composition:=True` and not `create_own_container`. Load as a composable node but delegate to the user to specify the ROS container with the `container_name` parameter.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

# 🎉 New feature

Closes #<NUMBER>

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

# ➡️ Forward port

Port <FROM_BRANCH> to <TO_BRANCH>

Branch comparison: https://github.com/gazebosim/<REPO>/compare/<TO_BRANCH>...<FROM_BRANCH>

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

# 🎈 Release

Preparation for <X.Y.Z> release.

Comparison to <x.y.z>: https://github.com/gazebosim/<REPO>/compare/<LATEST_TAG_BRANCH>...<RELEASE_BRANCH>

Needed by <PR(s)>

## Checklist
- [ ] Asked team if this is a good time for a release
- [ ] There are no changes to be ported from the previous major version
- [ ] No PRs targeted at this major version are close to getting in
- [ ] Bumped minor for new features, patch for bug fixes
- [ ] Updated changelog
- [ ] Updated migration guide (as needed)
- [ ] Link to PR updating dependency versions in appropriate repository in [gazebo-release](https://github.com/gazebo-release) (as needed): <LINK>

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: ros_gz_bridge/launch/ros_gz_bridge.launch.py [hunk 1]
```diff
@@ -46,6 +47,12 @@ def generate_launch_description():
         description='Name of container that nodes will load in if use composition',
     )
 
+    declare_start_container_cmd = DeclareLaunchArgument(
+        'start_container',
```
Comment (azeey) [inline ros_gz_bridge/launch/ros_gz_bridge.launch.py]: I would suggest changing this to `create_own_container` to make it a little more clear that this is not necessary for the bridge to work as a composable node. I would also add more to the description so users don't think they have to set this to True in order to get the benefits of composition.

**Review by azeey — COMMENTED** [review]
I think we should also make `use_composition=False` the default. The scenario I'd like to prevent is that the Gazebo server is started using  `gz sim -s` (I.e, not using the GzServer launch action) and the bridge is started using the `RosGzBridge` action. Since the default now is `use_composition=True`, it will try to add itself into an existing container, but due to the way Gazebo was started, that container will not be available.

File: ros_gz_bridge/launch/ros_gz_bridge.launch.py (hunk 1 above)
Comment (caguero) [reply ros_gz_bridge/launch/ros_gz_bridge.launch.py]: I renamed the new parameter as suggested and added to the launch action. I also changed the default value of `use_composition` to `False`. [f0f63b3](https://github.com/gazebosim/ros_gz/pull/616/commits/f0f63b3d9182506687cd88a14bf93ea726afaec6)

File: ros_gz_bridge/launch/ros_gz_bridge.launch.py [hunk 2]
```diff
@@ -46,12 +47,18 @@ def generate_launch_description():
         description='Name of container that nodes will load in if use composition',
     )
 
+    declare_create_own_container_cmd = DeclareLaunchArgument(
+        'create_own_container',
+        default_value='False',
+        description='Whether to start a ROS container when using composition',
```
Comment (azeey) [inline ros_gz_bridge/launch/ros_gz_bridge.launch.py]: I haven't checked if the formatting is okay, but I suggest adding more documentation here and saying this is not recommended.

```suggestion
        description='Whether the bridge should starts its own ROS container when using composition (Not Recommended). \
         This option should only be set to true if the you plan to put your ROS node in the container created by the bridge. \
          This is not needed if you want Gazebo and the bridge to be in the same ROS container. ',
```

**Comment by azeey** [comment]
Should we do the same for the gz_server launch file? Right now, it's not possible to put the GzServer node in a container created by the consumer.

**Comment by caguero** [comment]
> Should we do the same for the gz_server launch file? Right now, it's not possible to put the GzServer node in a container created by the consumer.

I think so. 

What do you think about the default value for `use_composition` in the gz_server launch? In some way it's more of an end-application, which makes me think a value of `True` isn't as problematic here. But at the same time having a consistency set of launch files (defaulting `use_composition` to False like in the bridge) makes it easier to use.

**Comment by azeey** [comment]
I don't have a strong opinion, but I lean toward making `use_composition=False` for both launch files by default. It feels safer at this point.

**Comment by caguero** [comment]
> I don't have a strong opinion, but I lean toward making `use_composition=False` for both launch files by default. It feels safer at this point.

Thanks, changed to `use_composition:=False` and `create_own_container:=False` by default.

**Comment by ahcorde** [comment]
@ros-pull-request-builder retest this please

**Review by ahcorde — APPROVED** [review]

**Comment by caguero** [comment]
https://github.com/Mergifyio backport jazzy

**Pull request merged**